### ISSUE TITLE
Return value section loses text in backticks

### ISSUE THREAD

**Issue by wch** [machow/quartodoc#96]
For example, the input text in the docstring is this (https://github.com/rstudio/py-shiny/blob/8cafc4470691961a8f994bcfc819d13d19f9deed/shiny/ui/_notification.py#L54-L56):

```
    Returns
    -------
    The notification's ``id``.
```

However, the output qmd looks like this:

```

## Returns

| Type   | Description        |
|--------|--------------------|
| The notification's  | |

```

And the HTML:

<img width="784" alt="image" src="https://user-images.githubusercontent.com/86978/231351042-2d6a9673-e9a6-413f-bc69-de02c31f7cae.png">

I tried changing the double backticks around `id` to single backticks, but it made no difference.

**Comment by machow** [comment]
I don't think the returns section conforms to numpydoc style. It looks like the parser thinks the description is type information.

numpydoc has two styles for return sections:

```
Returns
-------
int
    Description of anonymous integer return value.
```


```
Returns
-------
err_code : int
    Non-zero value indicates error code, or zero on success.
err_msg : str or None
    Human readable error message, or None on success.
```

https://numpydoc.readthedocs.io/en/latest/format.html#returns

**Comment by wch** [comment]
In all of our code, the functions have a return type annotation, like this:

```py
def f(x: int) -> str:
    """
    <docstring text here>
    
    Returns
    --------
    A string representation of `x`.
    """
     ...
```

Can quartodoc use that return type information? 

Perhaps the resulting markdown could look something like this:

```
## Returns

**str**: A string representation of `x`.

```

******

With the numpydoc's second style of return documentation, does that mean the returned value is a tuple? In practice, I think that is very uncommon, at least in our code. For us, if we return a tuple, I think it makes sense to document that the return value is a tuple with specified types, instead of using a table.

So if we had a function like this:

```py
def g(x: int) -> tuple(int, str | None):
    """
    <docstring text here>

    Returns
    --------
    A tuple where the first value is a return code, and the second value is an error message, or `None` if no error.
    """
     ...
```

The resulting markdown could be like this:

```
## Returns

**tuple(int, str | None)**: A tuple where the first value is a return code, and the second value is an error message, or `None` if no error.
```

**Comment by machow** [comment]
quartodoc uses numpydoc style, so if numpydoc supports it, then we should support it too. 

For the example you gave, it seems like you are using what numpydoc parses as the `type` field.

```python
from numpydoc.docscrape import NumpyDocString
ds = NumpyDocString(
    """
    <docstring text here>

    Returns
    --------
    A tuple where the first value is a return code, and the second value is an error message, or `None` if no error.
    """
)

ds["Returns"]
```

```
[Parameter(name='', type='A tuple where the first value is a return code, and the second value is an error message, or `None` if no error.', desc=[])]
```

**Comment by wch** [comment]
From #90 (but that comment was really about this issue):

> Why do you want backticks in the type column? I don't think there is a way to have a valid type annotation with backticks

I don't want backticks in the type column. I actually don't want a table, but even when I provided a customized `_render` function that puts the text in the second column, the code still disappears.

For example, see https://shiny.rstudio.com/py/api/ui.notification_show.html. This should say "The notification's `id`", but the `id` is not there.

<img width="831" alt="image" src="https://user-images.githubusercontent.com/86978/231885587-55477d9c-877a-483f-8680-5a8a5858173c.png">

**Comment by machow** [comment]
## tl;dr does this syntax work for you?

Using this syntax seems to work with the griffe parser (cf https://github.com/mkdocstrings/griffe/issues/137; please raise an issue there if you are concerned about stability of the behavior):

```
"""
Returns
--------
:
    Some description
"""
```

### Example

**file**: test_func.py

```python
def f(x) -> int:
    """
    Returns
    -------
    :
        the description
    """
```

```python
preview(get_object("test_func.f").docstring)
```

```
█─Docstring
├─parser = <Parser.numpy: 'numpy'>
└─parsed = █─list
           └─0 = █─DocstringSectionReturns
                 ├─kind = <DocstringSectionKind.returns: 'returns'>
                 ├─title = None
                 └─value = █─list
                           └─0 = █─DocstringReturn      # NOTE THIS RETURNS SECTION -----------------
                                 ├─name = ''
                                 ├─annotation = Name(source='int', full='int')
                                 └─description = 'the description'
```

## Background

It sounds like there are two issues at play:

1. **numpydoc style**: you don't want to have to specify type in the returns section (since the return annotation specifies it)
2. **griffe parser**: things in backticks in the return type annotation get dropped. seems less like a problem.

It looks like griffe handles automatically inferring the return type, when you just use "some_name: " in returns with no type on the right-hand side. I don't know what

```python
from griffe.dataclasses import Docstring
from griffe.docstrings.parsers import parse_numpy
from quartodoc import preview

doc = Docstring("""
    Returns
    -------
    :
        some `backtick` thing
    """
)
preview(parse_numpy(doc))
```

```
█─list
└─0 = █─DocstringSectionReturns
      ├─kind = <DocstringSectionKind.returns: 'returns'>
      ├─title = None
      └─value = █─list
                └─0 = █─DocstringReturn
                      ├─name = ''
                      ├─annotation = 'a'
                      └─description = 'some `backtick` thing'
```

**Comment by jcheng5** [comment]
@machow I just noticed that the Return value here is truncated: https://shiny.rstudio.com/py/api/render.plot.html (compare to the [docstring](https://github.com/rstudio/py-shiny/blob/95124eec149b691e2f4166086cb2f2ff3b8a51a7/shiny/render/_render.py#L332)

Should we be going through the py-shiny repo and changing all the docstrings to have the dummy `:` for the Returns section?

**Comment by machow** [comment]
Yeah -- writing the docstrings like this should fix it, and pull in the correct return type:

```python
def f(x) -> int:
    """
    Returns
    -------
    :
        the description
    """
```

**Comment by wch** [comment]
Update: I've updated the docstrings in shiny to use `:` as the return type.
